### PR TITLE
[CDAP-8336] Fix MetricsProcessorServiceTest with Kafka publish retry

### PR DESCRIPTION
**Pull request by ecfm**
https://issues.cask.co/browse/CDAP-8336
http://builds.cask.co/browse/CDAP-RUT525-9

### CODE REVIEW

File: cdap-watchdog/src/test/java/co/cask/cdap/metrics/process/MetricsProcessorServiceTest.java [hunk 1]
```diff
@@ -132,15 +129,15 @@ public void testMetricsProcessor() throws Exception {
                                            partitions, messagingService, injector.getInstance(SchemaGenerator.class),
                                            injector.getInstance(DatumReaderFactory.class),
                                            metricStore,
-                                           1);
+                                           1, 50);
     messagingMetricsProcessorService.startAndWait();
 
     // Publish metrics with messaging service and record expected metrics
     for (int i = 10; i < 20; i++) {
       publishMessagingMetrics(i, METRICS_CONTEXT, expected, SYSTEM_METRIC_PREFIX, MetricType.COUNTER);
```
Comment (sreevatsanraman) [inline cdap-watchdog/src/test/java/co/cask/cdap/metrics/process/MetricsProcessorServiceTest.java:137]: why is this part of code publishing metrics differently?

Comment (ecfm) [reply cdap-watchdog/src/test/java/co/cask/cdap/metrics/process/MetricsProcessorServiceTest.java:137]: This part is publishing metrics to TMS, while the previous part is publishing metrics to Kafka.

Comment (sreevatsanraman) [reply cdap-watchdog/src/test/java/co/cask/cdap/metrics/process/MetricsProcessorServiceTest.java:137]: ok

**Comment by sreevatsanraman** [comment]
LGTM

**Pull request merged**